ecsmanage/management/commands/ecsmanage.py

```diff
@@ -0,0 +1,210 @@
+
+        return config
+
+    def parse_response(self, response, key, idx=None):
+        """
+        Perform a key-value lookup on a response from the AWS API, wrapping it
+        in error handling such that if the lookup fails the response body
+        will get propagated to the end user.
+        """
+        if not response.get(key):
+            msg = f'Unexpected response from ECS API: {response}'
+            raise KeyError(msg)
+        else:
+            if idx is not None:
+                try:
+                    return response[key][0]
+                except (IndexError, TypeError):
+                    msg = (
+                        f"Unexpected value for '{key}' in response: "
+                        f'{response}'
+                    )
+                    raise IndexError(msg)
+            else:
+                return response[key]
+
+    def get_task_def(self, task_def_name):
+        """
+        Get the ARN of the latest ECS task definition for the app CLI.
+        """
+        task_def_response = self.ecs_client.list_task_definitions(
+            familyPrefix=task_def_name,
+            sort='DESC',
+            maxResults=1
+        )
+
+        return self.parse_response(task_def_response, 'taskDefinitionArns', 0)
+
+    def get_security_group(self, security_group_tags):
+        """
+        Get the ID of the security group to use for the app CLI.
+        """
+        filters = []
+        for tagname, tagvalue in security_group_tags.items():
+            filters.append({
+                'Name': f'tag:{tagname}',
+                'Values': [tagvalue]
+            })
+
+        sg_response = self.ec2_client.describe_security_groups(Filters=filters)
+
+        security_group = self.parse_response(sg_response, 'SecurityGroups', 0)
+        return security_group['GroupId']
+
+    def get_subnet(self, subnet_tags):
+        """
+        Get a subnet ID to use for the app CLI.
+        """
+        filters = []
+        for tagname, tagvalue in subnet_tags.items():
+            filters.append({
+                'Name': f'tag:{tagname}',
+                'Values': [tagvalue]
+            })
+
+        subnet_response = self.ec2_client.describe_subnets(Filters=filters)
+
+        subnet = self.parse_response(subnet_response, 'Subnets', 0)
+        return subnet['SubnetId']
+
+    def run_task(self,
+                 config,
+                 task_def_arn,
+                 security_group_id,
+                 subnet_id,
+                 cmd):
+        """
+        Run a task for a given task definition ARN using the given security
+        group and subnets, and return the task ID.
+        """
+        overrides = {
+            'containerOverrides': [
+                {
+                    'name': 'django',
+                    'command': cmd
+                }
+            ]
+        }
+
+        network_configuration = {
+            'awsvpcConfiguration': {
```

Comment:
Note that this assumes the user is running a task with the `awsvpc` network configuration. I don't have a good enough handle on AWS networking to know offhand the implications of this assumption.

Reply:
The `awsvpc` network configuration was introduced with Fargate. Fargate requires using the `awsvpc` network configuration.

Although the `awsvpc` network configuration is supported with the EC2 launch mode, there are special [considerations](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-networking.html#task-networking-considerations), and it's relatively unorthodox. 

If someone was using a traditional ECS cluster with EC2 container instances, this could be problematic. Could we just assume the network configuration from the existing task definition?

Reply:
Looks like the output of [`describe_task_definition`](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ecs.html#ECS.Client.describe_task_definition) includes a `compatibilities` attribute that specifies EC2 vs. Fargate, but as far as I can tell it doesn't return a specific network configuration. I'll look into whether there's a minimal EC2 networking config that we could use as a prefab option.

Reply:
Looking at the [`create_service`](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ecs.html#ECS.Client.create_service) docs:

>* **networkConfiguration** (_dict_) --
>The network configuration for the service. This parameter is required for task definitions that use the `awsvpc` network mode to receive their own elastic network interface, and it is not supported for other network modes.

Maybe we can just say, if the compatibilities don't include Fargate, don't include this block. It doesn't look like any of the network modes require `networkConfiguration`.

Looking at the [`run_task`](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ecs.html#ECS.Client.run_task) docs, if we're not using `awsvpc`, we shouldn't have to specify anything networking-related.

Reply:
By the way Jean, I really appreciate you looking into this. I also apologize that my review has been so delayed. A lot of things came up that were out-of-band and got pulled into this sprint. 

We weren't sure if you'd be able to prioritize resolving feedback in this PR, so I started [`feature/jrb/init-package`](https://github.com/azavea/django-ecsmanage/tree/feature/jrb/init-package) and [`feature/jrb/test-ecsmanage-package`](https://github.com/open-apparel-registry/open-apparel-registry/tree/feature/jrb/test-ecsmanage-package) where I was going to incorporate feedback myself, and then request a review from Hector. 

If you feel up to incorporating the changes related to `networkConfiguration`, I would rebase this PR against `feature/jrb/init-package`. There wasn't much else I was going to add, so after those changes are in, I can assign myself to this PR, and we can wait for Hector's review.

Reply:
Opening a new PR and assigning yourself is totally fine by me! I won't have time to look at this before Monday, so whatever you need to do to get it past the finish line is perfectly OK.